root/tag/TagIndex.js

```diff
@@ -0,0 +1,94 @@
+/*
+ * @flow
+ * Copyright (C) 2018 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import * as React from 'react';
+
+import ENTITIES from '../../entities';
+import DescriptiveLink from '../static/scripts/common/components/DescriptiveLink';
+import {l, ln, N_ln} from '../static/scripts/common/i18n';
+import NopArgs from '../static/scripts/common/i18n/NopArgs';
+
+import TagLayout from './TagLayout';
+
+type Props = {|
+  +tag: string,
+  +taggedEntities: {
+    +[string]: {|
+      +count: number,
+      +tags: $ReadOnlyArray<{|
+        +count: number,
+        +entity: CoreEntityT,
+        +entity_id: number,
+      |}>,
+    |},
+  },
+|};
+
+function buildSection<T>(
+  props: Props,
+  entityType: string,
+  title: string,
+  seeAllMessage: NopArgs,
+) {
+  const tags = props.taggedEntities[entityType];
+
+  if (!tags.count) {
+    return null;
+  }
+
+  const url = ENTITIES[entityType].url;
+
+  return (
+    <React.Fragment key={entityType}>
+      <h2>{title}</h2>
+      <ul>
+        {tags.tags.map(tag => (
+          <li key={tag.entity_id}>
+            <DescriptiveLink entity={tag.entity} />
+          </li>
+        ))}
+        {tags.count > tags.tags.length ? (
+          <li>
```

Comment:
Key attribute would probably be welcome although not absolutely needed here:
```suggestion
           <li key="count">
```